sdks/go/pkg/beam/util/grpcx/dial.go

```diff
@@ -23,9 +23,12 @@ import (
 	"google.golang.org/grpc"
 )
 
-// Dial is a convenience wrapper over grpc.Dial that specifies an insecure,
-// blocking connection with a timeout.
-func Dial(ctx context.Context, endpoint string, timeout time.Duration) (*grpc.ClientConn, error) {
+// Dial is a convenience wrapper over grpc.Dial. It can be overridden
+// to provide a customized dialing behavior.
+var Dial = DefaultDial
+
+// DefaultDial is a dialer that specifies an insecure blocking connection with a timeout.
+func DefaultDial(ctx context.Context, endpoint string, timeout time.Duration) (*grpc.ClientConn, error) {
```

Comment:
nit: name it InsecureDial instead.

Reply:
The pattern of X and DefaultX is used in the standard libraries, and I would prefer to adhere to this connection. Our default behavior is insecure, but that's not the defining trait. We also make a number of other policy decisions here (mostly omitting things), so default better captures the intent of this function.